src/test/java/com/googleinterns/zoomtube/servlets/DiscussionServletTest.java

```diff
@@ -83,11 +83,62 @@ public void doPost_loggedOut_postForbidden() throws ServletException, IOExceptio
 
     servlet.doPost(request, response);
 
-    verify(response).sendError(HttpServletResponse.SC_FORBIDDEN, "You are not logged in.");
+    verify(response).sendError(
+        HttpServletResponse.SC_FORBIDDEN, DiscussionServlet.ERROR_NOT_LOGGED_IN);
     assertThat(datastore.prepare(new Query(CommentUtil.KIND)).countEntities(withLimit(1)))
         .isEqualTo(0);
   }
 
+  @Test
+  public void doPost_missingLecture_badRequest() throws ServletException, IOException {
+    testServices.setEnvIsLoggedIn(true);
+
+    servlet.doPost(request, response);
+
+    verify(response).sendError(
+        HttpServletResponse.SC_BAD_REQUEST, DiscussionServlet.ERROR_MISSING_LECTURE);
+  }
+
+  @Test
+  public void doPost_missingType_badRequest() throws ServletException, IOException {
+    testServices.setEnvIsLoggedIn(true);
+    when(request.getParameter(DiscussionServlet.PARAM_LECTURE)).thenReturn(LECTURE_ID_STR);
+    when(request.getReader()).thenReturn(new BufferedReader(new StringReader("Untested Content")));
```

Comment:
"Untested Content" is used pretty frequently, consider declaring it global

Reply:
I wouldn't do that as mentioned in the "Local Variables and Class-Wide Constants" section of the unit test tech resources (https://docs.google.com/document/d/18T9ef-DRqTncVD377GoIQmOGI84cNKkZ7Nn07YURFTw/edit#heading=h.x20ltbjso2oq)

Reply:
So instead of making "Untested Content" a class-wide const I would make the BufferReader a class-wide variable. 

For simple types prefer raw values (ie, 1, null, Optional.empty(), "string"), but this BufferReader is not that simple so it can become a class-wide variable.